src/annotated.rs

```diff
@@ -1,47 +1,71 @@
-use std::iter::{Iterator, Peekable};
-use std::str::CharIndices;
+use std::iter::{Iterator};
 
 pub struct PartIterator<'a> {
-    string: &'a str,
-    char_indices: Peekable<CharIndices<'a>>,
-    boundary: usize,
+    s: &'a str,
+    start: usize,
+    white: bool,
 }
 
 impl<'a> PartIterator<'a> {
-    pub fn new(string: &'a str) -> PartIterator<'a> {
+    pub fn new(s: &'a str) -> PartIterator<'a> {
         PartIterator {
-            string,
-            char_indices: string.char_indices().peekable(),
-            boundary: 0,
+            s,
+            start: 0,
+            // Seems like there ought to be a way to get the first UTF-8
```

Comment:
I think you could get away with `s.starts_with(char::is_whitespace)`.

Reply:
So you'd be leaving off the leading `Part` for leading whitespace?

Reply:
I didn't mean to suggest any changes to the way the code works, just rewriting this code that you weren't happy with because it alludes to so many extraneous concepts.  I assumed you were looking for a way to write this that didn't require iterators, options, pattern matching.

Reply:
Yep, I totally misunderstood you and that would totally do it!